### PR TITLE
Add `vue/max-lines-per-block` rule

### LINKED ISSUE #1443: want: vue/max-line rule, limit the number of lines of code per block
**Please describe what the rule should do:**

This rule applies to the number of lines of code.

Help people not to write thousands of lines of javascript code in the same file. Split the code to improve maintainability and readability.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
  <view>
  <view>
</template>

<script>
// There is a very long javascript code here
export default {
 data () {
  return {}
 }
}
</script>

<style>
root {
  long-long-style: none;
}
</style>
```

**Additional context**

the rule look like:

```js
rules: {
  "vue/max-line": ['error', {
    // Specify a block of code via lang="xxx"
    // eg.
    // <style lang="sass"><style>
    // sass: 800
 	js: 800,         // limit max line for javascript block
    jsx: 800,
    ts: 800,         // limit max line for typescript block
    tsx: 800,
    html: 1200,      // limit max line for HTML block
    css: 1200        // limit max line for css/sass/scss block
  }]
}
```

It receives a parameter to specify the maximum length of a certain code block

### PR DESCRIPTION
**Pull request by lsdsjy**
Closes #1443

### CODE REVIEW

File: docs/rules/max-lines-per-block.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+This rule enforces a maximum number of lines per block, in order to aid in maintainability and reduce complexity.
+
+## :wrench: Options
+
+This rule takes an object, where you can specify the maximum number of lines in each type of SFC block and customize the line counting behavior.
+The following properties can be specified for the object.
+
+- `script` ... Specify the maximum number of lines in &lt;script&gt; block. Won't enforce limitation if not specified.
+- `template` ... Specify the maximum number of lines in &lt;template&gt; block. Won't enforce limitation if not specified.
+- `style` ... Specify the maximum number of lines in &lt;style&gt; block. Won't enforce limitation if not specified.
```
Comment (FloEdelmann) [inline docs/rules/max-lines-per-block.md:24]:
```suggestion
- `script` ... Specify the maximum number of lines in `<script>` block. Won't enforce limitation if not specified.
- `template` ... Specify the maximum number of lines in `<template>` block. Won't enforce limitation if not specified.
- `style` ... Specify the maximum number of lines in `<style>` block. Won't enforce limitation if not specified.
```

File: docs/rules/max-lines-per-block.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+<eslint-code-block :rules="{'vue/max-lines-per-block': ['error', { script: 1, skipBlankLines: true }]}">
+
+```vue
+<!-- ✓ GOOD -->
+<script>
+
+  console.log(1)
+</script>
+```
```
Comment (FloEdelmann) [inline docs/rules/max-lines-per-block.md:48]: We need to show which options are used in each example as headers:

````suggestion
### `{ template: 2 }`

<eslint-code-block :rules="{'vue/max-lines-per-block': ['error', { template: 2 }]}">

```vue
<!-- ✗ BAD -->
<template>
  <div>
    hi
  </div>
</template>
```

</eslint-code-block>

### `{ script: 1, skipBlankLines: true }`

<eslint-code-block :rules="{'vue/max-lines-per-block': ['error', { script: 1, skipBlankLines: true }]}">

```vue
<!-- ✓ GOOD -->
<script>

  console.log(1)
</script>
```
````

**Review by FloEdelmann — CHANGES_REQUESTED** [review]
Looks good, thanks for contributing!

I have just two small suggestions.

**Review by FloEdelmann — APPROVED** [review]
LGTM :slightly_smiling_face:

**Review by ota-meshi — APPROVED** [review]
Thank you!

**Pull request merged**